### PR TITLE
Add sync Android SDK

### PR DESCRIPTION
**Pull request by danielpassos**

### CODE REVIEW

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -60,6 +60,7 @@ include::{partialsdir}/attributes.adoc[]
 *** xref:data_sync.adoc#introduction[Introduction]
 *** xref:data_sync.adoc#setup[Setting Up]
 *** xref:data_sync.adoc#ui[The Data Sync UI]
+*** xref:data_sync.adoc#sdk[Addind sync to your app]
```
Comment (darahayes) [inline modules/ROOT/nav.adoc]: Addind -> Adding

File: modules/ROOT/pages/sync/adding-sync-to-your-app.adoc [hunk 2]
```diff
@@ -9,6 +9,94 @@ This section will help you to set up the {sync-service} service SDK in your App.
...
+****
+
+[role="secondary"]
+.Xamarin
+****
+****
+
+## Using the Sync Mobile Service
+
+Client SDK provides a simple way to get the Apollo client configured to use.
```
Comment (darahayes) [inline modules/ROOT/pages/sync/adding-sync-to-your-app.adoc]: I would maybe change this to something like 'The Sync SDK provides a method to get a configured instance of the Apollo Client'

**Comment by darahayes** [comment]
@danielpassos thanks for the PR. I have left some very minor feedback.

File: modules/ROOT/pages/sync/adding-sync-to-your-app.adoc [hunk 3]
```diff
@@ -9,6 +9,94 @@ This section will help you to set up the {sync-service} service SDK in your App.
...
+
+[role="primary"]
+.Android
+****
+```
+SyncService
+        .getInstance()
+        .getApolloClient()
+        .query(MyNotesQuery.builder().build())
+        .enqueue(new ApolloCall.Callback<MyNotesQuery.Data>() {
```
Comment (wtrocki) [inline modules/ROOT/pages/sync/adding-sync-to-your-app.adoc]: People will get error on MyNotesQuery object if they not follow schema above. Best to mention that `MyNotes` is generated from example schema above (in the note)

File: modules/ROOT/pages/sync/adding-sync-to-your-app.adoc [hunk 4]
```diff
@@ -9,6 +9,94 @@ This section will help you to set up the {sync-service} service SDK in your App.
...
+The Sync SDK is using the link:https://github.com/apollographql/[Apollo GraphQL Client] behind of the scenes to handle link:http://graphql.github.io/[GraphQL]. 
+
+link:https://github.com/apollographql[Apollo client] requires some configuration to work.
+
+[role="primary"]
+.Android
+****
+1. Create a special folder `graphql` inside of `src/main`
+2. Create a file `schema.json` with your GraphQL schema
+3. Create a file with name that corresponds objects you want to create and the `.graphql` extension
```
[the remark this answers is not in the document]
Comment (danielpassos) [reply modules/ROOT/pages/sync/adding-sync-to-your-app.adoc]: Well, that file is not for _query_ only

[the remark this answers is not in the document]
Comment (pb82) [reply modules/ROOT/pages/sync/adding-sync-to-your-app.adoc]: Ok, then how about: `Create a file with a name that corresponds to your business objects and a .graphql extension`?

**Comment by danielpassos** [comment]
@da

**Comment by danielpassos** [comment]
@darahayes, @wtrocki, @pb82 second 👁 ?

File: modules/ROOT/pages/sync/adding-sync-to-your-app.adoc [hunk 5]
```diff
@@ -9,6 +9,96 @@ This section will help you to set up the {sync-service} service SDK in your App.
...
+
+```
+query myNotes {
+  allNotes() {
+    id
+  }
+}
+```
+
+Apollo will automatically create a file `MyNotesQuery` (in compile time) to you fire your query.
```
Comment (pb82) [inline modules/ROOT/pages/sync/adding-sync-to-your-app.adoc]: `to you fire your query` -> `for you to fire the query`

**Pull request merged**